pkg/solana/chain.go

```diff
@@ -164,12 +171,15 @@ func (v *verifiedCachedClient) verifyChainID(ctx context.Context) (bool, error)
 		return v.chainIDVerified, fmt.Errorf("failed to fetch ChainID in verifiedCachedClient: %w", err)
 	}
 
-	// if this is localnet, allow any chain ID as long as it's not spoofing an official network
-	if v.expectedChainID != "localnet" {
+	// Check if configured chain ID could be a genesis hash
```

Comment:
instead of checking if it `could be` a genesis hash, would be useful to check if it actually matches one of the `chainIDs` we have registered? Checking provided chainID against [this map](https://github.com/smartcontractkit/chain-selectors/blob/239b77a97d483b3fc4daf39beff47255b936e2be/solana.go#L70-L76) could work

lmk if I'm way off here, just saw the PR but don't have much context about it

Reply:
Nice catch, we have a thread on this and decided to use `GetChainDetailsByChainIDAndFamily`